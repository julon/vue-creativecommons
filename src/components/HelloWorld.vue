<template lang="pug">
  .hello
    h1 {{ msg }}
    .cat(v-for="category in categories")
      h2 {{ category.title }}
      ul
        li(v-for="link in category.links")
          a(:href="link.href" target="_blank") {{ link.label }}
</template>

<script>
export default {
  name: "HelloWorld",
  data() {
    return {
      msg: "Welcome to Your Vue.js Library",
      categories: [
        {
          title: "Essential Links",
          links: [
            { label: "Core Docs", href: "https://vuejs.org" },
            { label: "Forum", href: "https://forum.vuejs.org" },
            { label: "Community Chat", href: "https://chat.vuejs.org" },
            { label: "Twitter", href: "https://twitter.com/vuejs" },
            {
              label: "Docs for This Template",
              href: "http://vuejs-templates.github.io/webpack/"
            }
          ]
        },
        {
          title: "Ecosystem",
          links: [
            { label: "vue-router", href: "https://vuejs.org" },
            { label: "vuex", href: "https://forum.vuejs.org" },
            { label: "vue-loader", href: "https://chat.vuejs.org" },
            { label: "awesome-vue", href: "https://twitter.com/vuejs" }
          ]
        }
      ]
    };
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="stylus" scoped>
h1, h2
  font-weight: normal
ul
  list-style-type: none
  padding: 0
li
  display: inline-block
  margin: 0 10px
a
  color: #42b983
</style>
