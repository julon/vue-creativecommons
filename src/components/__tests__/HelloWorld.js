import Vue from "vue";
import HelloWorld from "../HelloWorld";

describe("Hello.vue", () => {
  const Constructor = Vue.extend(HelloWorld);
  const vm = new Constructor().$mount();
  test("should render correct contents", () => {
    expect(vm.$el.querySelector(".hello h1").textContent).toEqual(
      "Welcome to Your Vue.js Library"
    );
  });
  test("should match the snapshot", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
