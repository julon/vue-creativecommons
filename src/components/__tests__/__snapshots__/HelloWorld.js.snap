// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue should match the snapshot 1`] = `
<div
  class="hello"
>
  <h1>
    Welcome to Your Vue.js Library
  </h1>
  <div
    class="cat"
  >
    <h2>
      Essential Links
    </h2>
    <ul>
      <li>
        <a
          href="https://vuejs.org"
          target="_blank"
        >
          Core Docs
        </a>
      </li>
      <li>
        <a
          href="https://forum.vuejs.org"
          target="_blank"
        >
          Forum
        </a>
      </li>
      <li>
        <a
          href="https://chat.vuejs.org"
          target="_blank"
        >
          Community Chat
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/vuejs"
          target="_blank"
        >
          Twitter
        </a>
      </li>
      <li>
        <a
          href="http://vuejs-templates.github.io/webpack/"
          target="_blank"
        >
          Docs for This Template
        </a>
      </li>
    </ul>
  </div>
  <div
    class="cat"
  >
    <h2>
      Ecosystem
    </h2>
    <ul>
      <li>
        <a
          href="https://vuejs.org"
          target="_blank"
        >
          vue-router
        </a>
      </li>
      <li>
        <a
          href="https://forum.vuejs.org"
          target="_blank"
        >
          vuex
        </a>
      </li>
      <li>
        <a
          href="https://chat.vuejs.org"
          target="_blank"
        >
          vue-loader
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/vuejs"
          target="_blank"
        >
          awesome-vue
        </a>
      </li>
    </ul>
  </div>
</div>
`;
